Add into the github template a "Docs" section (#558)
SUSE/skuba: .github/pull_request_template.md

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

> If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.

Reviewer: ```suggestion
If docs need to be updated, please create an issue in https://github.com/SUSE/doc-caasp and link it in this PR.
```

Author: If we ask the developers to create an issue ... they won't create a PR. The idea is that the developer provides the docs, as they have been doing, as PR for you to review. If they create only the issue, then you have to create the docs and, given we have more devs than writers, this will be a bottleneck.

Reviewer: Right. I didn't understand that at first, but it makes sense. It will just be up to the documentarian to then create an issue, so that we can allot time for reviewing or otherwise editing the PR.

> At the time of creating this issue, This PR can be work in progress (set its title to [WIP]...),

Reviewer: ```suggestion
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
```

> but it needs to be ready in order to merge this PR.

Reviewer: ```suggestion
but the documentation needs to be finalized before the PR can be merged. 
```
